pkg/network/tracer/tracer_linux_test.go

```diff
@@ -1724,107 +1724,69 @@ func (s *TracerSuite) TestBlockingReadCounts() {
 	assert.Equal(t, uint64(n), conn.Monotonic.RecvBytes)
 }
 
-func (s *TracerSuite) TestTCPDirectionWithPreexistingConnection() {
-	t := s.T()
-	wg := sync.WaitGroup{}
-
-	// setup server to listen on a port
-	server := NewTCPServer(func(c net.Conn) {
-		t.Logf("received connection from %s", c.RemoteAddr())
-		_, err := bufio.NewReader(c).ReadBytes('\n')
-		if err == nil {
-			wg.Done()
-		}
-	})
-	server.Run()
-	t.Cleanup(server.Shutdown)
-	t.Logf("server address: %s", server.address)
-
-	// create an initial client connection to the server
-	c, err := net.DialTimeout("tcp", server.address, 5*time.Second)
-	require.NoError(t, err)
-	t.Cleanup(func() { c.Close() })
-
-	// start tracer so it dumps port bindings
-	cfg := testConfig()
-	tr := setupTracer(t, cfg)
-
-	// open and close another client connection to force port binding delete
-	c2, err := net.DialTimeout("tcp", server.address, 5*time.Second)
-	require.NoError(t, err)
-	wg.Add(1)
-	_, err = c2.Write([]byte("conn2\n"))
-	require.NoError(t, err)
-	c2.Close()
-	wg.Wait()
-
-	wg.Add(1)
-	// write some data so tracer determines direction of this connection
-	_, err = c.Write([]byte("original\n"))
-	require.NoError(t, err)
-	wg.Wait()
-
-	var origConn []network.ConnectionStats
-	// the original connection should still be incoming for the server
-	require.Eventually(t, func() bool {
-		conns := getConnections(t, tr)
-		origConn = searchConnections(conns, func(cs network.ConnectionStats) bool {
-			return fmt.Sprintf("%s:%d", cs.Source, cs.SPort) == server.address &&
-				fmt.Sprintf("%s:%d", cs.Dest, cs.DPort) == c.LocalAddr().String()
-		})
-
-		return len(origConn) == 1
-	}, 3*time.Second, 500*time.Millisecond, "timed out waiting for original connection")
-
-	require.Equal(t, network.INCOMING, origConn[0].Direction, "original server<->client connection should have incoming direction")
-}
-
 func (s *TracerSuite) TestPreexistingConnectionDirection() {
 	t := s.T()
 	// Start the client and server before we enable the system probe to test that the tracer picks
 	// up the pre-existing connection
 
 	server := NewTCPServer(func(c net.Conn) {
 		r := bufio.NewReader(c)
-		_, _ = r.ReadBytes(byte('\n'))
-		_, _ = c.Write(genPayload(serverMessageSize))
-		_ = c.Close()
+		for {
+			if _, err := r.ReadBytes(byte('\n')); err == io.EOF {
+				return
+			}
```

Comment:
do we need to handle non EOF errors here?

Reply:
Yes, fixed.